src/Chapter1.hs

```diff
@@ -542,7 +546,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z 
+  | x <= y && y <= z || z <= y && y <= x = y
+  | x <= z && z <= y || y <= z && z <= x = z
+  | y <= x && x <= z || z <= x && x <= y = x
+  | otherwise = error "wat?"
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive` if you don't add `otherwise` case, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

However, in this case, you can just combine the previous one with the `otherwise` case, and write: 
```suggestion
  | otherwise = x
```